cmip6_downscaling/config/config.py

```diff
@@ -1,104 +1,130 @@
-        cpu_request=return_kubernetes_config()["kubernetes_cpu"],
-        memory_request=return_kubernetes_config()["kubernetes_memory"],
-        image=return_docker_config()["image"],
-        labels=return_prefect_config()["agent"],
-        env=return_env_config()["env_config"],
-    )
-    return kubernetes_run_config
-
-
-def return_dask_executor():
-
-    dask_executor = DaskExecutor(
-        cluster_class=lambda: KubeCluster(
-            make_pod_spec(
-                image=return_docker_config()["image"],
-                memory_limit=return_dask_config()["dask_executor_memory_limit"],
-                memory_request=return_dask_config()["dask_executor_memory_request"],
-                threads_per_worker=return_dask_config()["dask_executor_threads_per_worker"],
-                cpu_limit=return_dask_config()["dask_executor_cpu_limit]"],
-                cpu_request=return_dask_config()["dask_executor_cpu_request"],
-                env=return_env_config()["env_config"],
+class AbstractConfig:
+    # attributes
+    # prefect_storage
+    # prefect_run_config
+    # prefect_executor
+    # fsspec_fs
+    def __init__(self):
+        self.connection_string = os.environ.get("AZURE_STORAGE_CONNECTION_STRING")
+        self.agent = "az-eu-west"
+        self.storage = Azure("prefect")
+        self.extra_pip_packages = "git+https://github.com/carbonplan/cmip6-downscaling.git git+https://github.com/pangeo-data/scikit-downscale.git git+https://github.com/NCAR/xpersist.git"
+        self.intermediate_cache_path = "az://flow-outputs/intermediate"
+        self.results_cache_path = "az://flow-outputs/results"
+        self.serializer = "xarray.zarr"
+
+
+class CloudConfig(AbstractConfig):
+    def __init__(self):
+        super().__init__()
+        self.kubernetes_cpu = 7
+        self.kubernetes_memory = "16Gi"
+        self.image = "carbonplan/cmip6-downscaling-prefect:2021.12.06"
+        self.OMP_NUM_THREADS = "1"
+        self.MPI_NUM_THREADS = "1"
+        self.MKL_NUM_THREADS = "1"
+        self.OPENBLAS_NUM_THREADS = "1"
+        self.pod_memory_limit = "4Gi"
+        self.pod_memory_request = "4Gi"
+        self.pod_threads_per_worker = 2
+        self.pod_cpu_limit = 2
+        self.pod_cpu_request = 2
+        self.deploy_mode = "remote"
+        self.adapt_min = 2
+        self.adapt_max = 2
+        self.dask_distributed_worker_resources_taskslots = "1"
+
+    def generate_env(self):
+        env = {
+            "AZURE_STORAGE_CONNECTION_STRING": self.connection_string,
+            "EXTRA_PIP_PACKAGES": self.extra_pip_packages,
+            "OMP_NUM_THREADS": self.OMP_NUM_THREADS,
+            "MPI_NUM_THREADS": self.MPI_NUM_THREADS,
+            "MKL_NUM_THREADS": self.MKL_NUM_THREADS,
+            "OPENBLAS_NUM_THREADS": self.OPENBLAS_NUM_THREADS,
+            "DASK_DISTRIBUTED__WORKER__RESOURCES__TASKSLOTS": self.dask_distributed_worker_resources_taskslots,
+        }
+        return env
+
+    def dask_executor(self):
```

Comment:
You probably want to rename this to simply `executor` so each config can have the same method.